packages/web/src/pages/upload-page/fields/ReleaseDateField.tsx

```diff
@@ -1,94 +1,323 @@
+    'Specify a release date for your music or schedule it to be released in the future. Release date affects sorting on your profile and is visible in track details.',
+  callout: (timePeriod: TimePeriodType) => {
+    if (timePeriod === TimePeriodType.PAST) {
+      return (
+        <>
+          Setting a release date in the past will impact the order tracks appear
+          on your profile.
+        </>
+      )
+    } else {
+      return (
+        <>
+          Your scheduled track will become live on Audius on the date and time
+          you’ve chosen above in your time zone.
+        </>
+      )
+    }
+  }
 }
 
 const RELEASE_DATE = 'release_date'
+const RELEASE_DATE_HOUR = 'release_date_hour'
+const RELEASE_DATE_MERIDIAN = 'release_date_meridian'
+const RELEASE_DATE_TYPE = 'release_date_type'
 
 export type ReleaseDateFormValues = {
   [RELEASE_DATE]: string | undefined
+  [RELEASE_DATE_HOUR]: string
+  [RELEASE_DATE_MERIDIAN]: string
+  [RELEASE_DATE_TYPE]: string
+}
+
+export enum ReleaseDateType {
+  RELEASE_NOW = 'RELEASE_NOW',
+  SCHEDULED_RELEASE = 'SCHEDULED_RELEASE'
+}
+
+export enum TimePeriodType {
+  PAST = 'PAST',
+  PRESENT = 'PRESENT',
+  FUTURE = 'FUTURE'
+}
+
+export enum ReleaseDateMeridian {
+  AM = 'AM',
+  PM = 'PM'
+}
+
+const timeValidationSchema = z.object({
+  release_date_hour: z
+    .string()
+    .refine((value) => /^([0-9]|0[1-9]|1[0-2]):([0-5][0-9])$/.test(value), {
+      message: 'Invalid time.'
+    })
+})
+
+const getScheduledReleaseLabelMessage = (
+  releaseDate: string | null,
+  prefixMessage = ''
+) => {
+  if (!releaseDate) {
+    return 'Today'
+  }
+  const formatReleaseMessage = (releaseDate: string, base: string) => {
+    const isFutureRelease = moment(releaseDate ?? undefined).isAfter(
+      moment.now()
+    )
+    let message = isFutureRelease ? '[' + prefixMessage + '] ' : ''
+    message += base
+    message += isFutureRelease ? ' @ LT' : ''
+    return message
+  }
+
+  return moment(releaseDate ?? undefined).calendar(undefined, {
+    sameDay: formatReleaseMessage(releaseDate, '[Today]'),
+    nextDay: formatReleaseMessage(releaseDate, '[Tomorrow]'),
+    nextWeek: formatReleaseMessage(releaseDate, 'dddd'),
+    lastDay: '[Yesterday]',
+    lastWeek: '[Last] dddd',
+    sameElse: formatReleaseMessage(releaseDate, 'MM/DD/YYYY')
+  })
 }
 
 type ReleaseDateValue = SingleTrackEditValues[typeof RELEASE_DATE]
 
 export const ReleaseDateField = () => {
-  const [{ value }, , { setValue }] =
+  const [trackReleaseDateField, , { setValue: setTrackReleaseDate }] =
     useTrackField<ReleaseDateValue>(RELEASE_DATE)
+  const trackReleaseDate = trackReleaseDateField.value
 
-  const initialValues = useMemo(
-    () => ({ [RELEASE_DATE]: value ?? undefined }),
-    [value]
-  )
+  const [releaseDateTypeField, ,] = useField(RELEASE_DATE_TYPE)
+  const releaseDateType = releaseDateTypeField.value
+  const roundUpHour = moment().add(1, 'hours').minutes(0).seconds(0)
```

Comment:
yea true it gets confusing. if you open the form at 2:59 PM, it'll round to 3 PM by default. this difference isn't much. it's the difference of having a scheduled release for 1 minute vs a release that's 1 minute later than intended. that's kind of on the user in that case.

with this other case, selecting 23:30 would round to midnight. so a scheduled release would default to today at 12 AM.

either case the user is aware of the release time, on the release date form and parent form's "scheduled for" label. but i agree it can lead to unexpected results, so i'm gonna remove this rounding up and we'll see how it works on stage.